### PR TITLE
refactor(git): Extract types to separate file

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Extract interfaces to `types.ts`
- Update depending imports
- Change `GitNoVerifyOption` type to `'push' | 'commit'` union

## Context:

- Ref: #7154

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Blocks #12842

**Review by rarkins — DISMISSED** [review]

File: lib/util/git/index.ts [hunk 1]
```diff
@@ -22,45 +15,27 @@ import {
...
+import { GitProtocol } from '../../types/git';
 import { Limit, incLimitedValue } from '../../workers/global/limits';
 import { regEx } from '../regex';
 import { parseGitAuthor } from './author';
-import { GitNoVerifyOption, getNoVerify, simpleGitConfig } from './config';
+import { getNoVerify, simpleGitConfig } from './config';
 import { configSigningKey, writePrivateKey } from './private-key';
-
-export { GitNoVerifyOption, setNoVerify } from './config';
+import {
```
Comment (viceice) [inline lib/util/git/index.ts:24]:
```suggestion
import type {
```

File: lib/util/git/index.ts [hunk 2]
```diff
@@ -22,45 +15,27 @@ import {
 } from '../../constants/error-messages';
 import { logger } from '../../logger';
 import { ExternalHostError } from '../../types/errors/external-host-error';
-import { GitOptions, GitProtocol } from '../../types/git';
+import { GitProtocol } from '../../types/git';
```
Comment (viceice) [inline lib/util/git/index.ts:18]:
```suggestion
import type { GitProtocol } from '../../types/git';
```

File: lib/util/git/index.ts [hunk 3]
```diff
@@ -1,13 +1,6 @@
...
-  Options,
-  ResetMode,
-  SimpleGit,
-  StatusResult as StatusResult_,
-  TaskOptions,
-} from 'simple-git';
+import Git, { Options, ResetMode, SimpleGit, TaskOptions } from 'simple-git';
 import { join } from 'upath';
 import { configFileNames } from '../../config/app-strings';
 import { GlobalConfig } from '../../config/global';
```
Comment (viceice) [inline lib/util/git/index.ts:6]:
```suggestion
import type { GlobalConfig } from '../../config/global';
```

File: lib/util/git/types.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import type {
+  DiffResult as DiffResult_,
+  StatusResult as StatusResult_,
+} from 'simple-git';
+import type { GitOptions } from '../../types/git';
+
+export type StatusResult = StatusResult_;
+export type DiffResult = DiffResult_;
```
Comment (viceice) [inline lib/util/git/types.ts:8]: Do we really need this? Can't we direct export those?

File: lib/workers/branch/index.spec.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ import * as _npmPostExtract from '../../manager/npm/post-update';
 import type { WriteExistingFilesResult } from '../../manager/npm/post-update/types';
 import { PrState } from '../../types';
 import * as _exec from '../../util/exec';
-import { File, StatusResult } from '../../util/git';
+import { File, StatusResult } from '../../util/git/types';
```
Comment (viceice) [inline lib/workers/branch/index.spec.ts:13]:
```suggestion
import type { File, StatusResult } from '../../util/git/types';
```

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/util/git/types.ts (hunk 4 above)
Comment (zharinov) [reply lib/util/git/types.ts:8]: Let's try...

Comment (viceice) [reply lib/util/git/types.ts]: I think this is a leftover from `isolatedModules:true`, which was required for babel

**Review by viceice — APPROVED** [review]

File: lib/util/git/types.ts (hunk 4 above)
Comment (zharinov) [reply lib/util/git/types.ts]: Yes, I don't know this story, but seemed pretty unnatural

**Pull request merged**